api/actions/v1alpha1/latencysampler_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/odigos-io/odigos/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LatencySamplerSpec defines the desired state of LatencySampler action
+type LatencySamplerSpec struct {
+	ActionName string                       `json:"actionName,omitempty"`
+	Notes      string                       `json:"notes,omitempty"`
+	Disabled   bool                         `json:"disabled,omitempty"`
+	Signals    []common.ObservabilitySignal `json:"signals"`
+
+	// Specifies the lower latency threshold in milliseconds; traces with latency equal to or exceeding this value will be considered for sampling.
+	// +kubebuilder:validation:Required
+	MinimumLatencyThreshold int `json:"minimum_latency_threshold"`
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
```

Comment:
I think its will be common for users to define multiple endpoints. If one action can only hold one endpoint, it will require for users to define, potentially very long list of actions which might not by idle.

Consider making each action contain a list of service-endpoints-threshold.